src/main/java/org/opensearch/neuralsearch/stats/info/InfoStatsManager.java

```diff
@@ -162,12 +199,66 @@ private void countTextChunkingProcessorStats(Map<InfoStatName, CountableInfoStat
         }
     }
 
+    /**
+     * Adds search processor info stats, mutating the input
+     * @param stats mutable map of info stats that the result will be added to
+     */
+    private void addSearchProcessorStats(Map<InfoStatName, CountableInfoStatSnapshot> stats) {
+        List<Map<String, Object>> pipelineConfigs = pipelineServiceUtil.getSearchPipelineConfigs();
+
+        // Iterate through all search processors and count their stats individually by calling helpers
+        for (Map<String, Object> pipelineConfig : pipelineConfigs) {
+            // Search phase results processors
+            List<Map<String, Object>> phaseResultsProcessors = asListOfMaps(pipelineConfig.get(PHASE_RESULTS_PROCESSORS_KEY));
+            for (Map<String, Object> phaseResultsProcessor : phaseResultsProcessors) {
+                for (Map.Entry<String, Object> entry : phaseResultsProcessor.entrySet()) {
+                    String processorType = entry.getKey();
+                    Map<String, Object> processorConfig = asMap(entry.getValue());
+                    switch (processorType) {
+                        case NormalizationProcessor.TYPE:
+                            countNormalizationProcessorStats(stats, processorConfig);
+                            break;
+                        case RRFProcessor.TYPE:
+                            countRRFProcessorStats(stats, processorConfig);
+                            break;
```

Comment:
please add default case and throw one of runtime exceptions in case we reach it

Reply:
Not sure if that makes sense here? Here we are iterating through all processors in the pipeline, which may contain processors from core or other plugins. If we encounter an `MLInferenceRequestProcessor` for example, we would not want to record stats for it, and  that would trigger the default case and throw an exception, breaking the API.